components/brave_ads/browser/brave_ads_driver.cc

```diff
@@ -0,0 +1,43 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_ads/browser/brave_ads_driver.h"
+
+#include <utility>
+
+#include "brave/components/brave_ads/browser/ads_service.h"
+#include "brave/components/brave_ads/browser/brave_ads_driver_delegate.h"
+#include "brave/components/brave_rewards/browser/rewards_service.h"
+
+namespace brave_ads {
+
+BraveAdsDriver::BraveAdsDriver(AdsService* ads_service,
```

Comment:
this class is never used by anything in components/brave_ads and this Driver/Delegate relationship seems to just add complexity for no benefit that I can see. Also `Driver` doesn't really make sense to me here. I think what you want is something more like `BraveSearchFallbackHost`

Reply:
The reason of `BraveAdsDriverDelegate` is to separate popup UI implementation for desktop and android.
`BraveAdsDriverDelegateImpl` - shows popup for desktops
`BraveAdsDriverDelegateImplAndroid` - shows popup for android
`BraveAdsDriver` contains logic that is common for both platforms

Maybe I can move it to `brave/browser/brave_ads/` ?

Reply:
Seems like common android and desktop implementations has very few common logic, so it is no need to create separate class for common logic. 
I will remove `BraveAdsDriverDelegate` and subclasses and will do only `BraveAdsDriver` and `BraveAdsDriverAndroid` classes

Reply:
Done